pkg/provision/primitives/counters.go

```diff
@@ -184,6 +186,35 @@ type resourceUnits struct {
 	CRU uint64 `json:"cru,omitempty"`
 }
 
+// CheckMemoryRequirements checks memory requirements for a reservation and compares it to whats in the counters
+// and what is the total memory on this node
+func (c *Counters) CheckMemoryRequirements(r *provision.Reservation, memStats *mem.VirtualMemoryStat) error {
+	switch r.Type {
+	case ContainerReservation:
+		requestedUnits, err := processContainer(r)
+		if err != nil {
+			return err
+		}
+
+		// If current MRU + requested MRU exceeds total, return an error
+		if uint64(c.CurrentUnits().Mru)+requestedUnits.HRU > memStats.Total {
+			return errors.New("reservation requested memory is to high")
+		}
+	case KubernetesReservation:
+		requestedUnits, err := processKubernetes(r)
+		if err != nil {
+			return err
+		}
+
+		// If current MRU + requested MRU exceeds total, return an error
+		if uint64(c.CurrentUnits().Mru)+requestedUnits.HRU > memStats.Total {
```

Comment:
this code is identical to the one for containers, instead can we have the switch set the "requestedUnits" value, and then handle the calculations after